driver_go18_test.go

```diff
@@ -196,3 +199,270 @@ func TestSkipResults(t *testing.T) {
 		}
 	})
 }
+
+func TestPingContext(t *testing.T) {
+	runTests(t, dsn, func(dbt *DBTest) {
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+		if err := dbt.db.PingContext(ctx); err != context.Canceled {
+			dbt.Errorf("expected context.Canceled, got %v", err)
+		}
+	})
+}
+
+var (
+	_ driver.ConnBeginTx        = &mysqlConn{}
```

Comment:
Please move all these static interface implementation checks (there are more below) to the top of the file